R/utils.R

```diff
@@ -412,9 +412,34 @@ capitalize <- function(x) {
 }
 
 # At some point around 2019, read_csv introduced new attributes that broke tests
+# NOTE: the first argument should be the path to the file to be read
 readr_read_csv <- function(...) {
-  data <- readr::read_csv(...)
+  # detect any non-delimited rows at the top of the file
+  skips <- detect_skip(..1)
+  # silence readr repairing names
+  data <- readr::read_csv(..., skip = skips, name_repair = "unique_quiet", show_col_types = FALSE)
+  # drop empty columns
+  data <- data[, vapply(data, function(x) !all(is.na(x)), TRUE)]
```

Comment:
This is because the new format has trailing commas in the data so `readr` thinks there's another column there.
```
Ch1Amplitude,Ch2Amplitude,1,2,
9265.351,781.6206,1,0,
```